third_party/amd/lib/TritonAMDGPUToLLVM/BufferOpsEmitter.cpp

```diff
@@ -161,10 +164,53 @@ void BufferEmitter::fillCommonArgs(Type type, Value rsrcDesc,
   Value sgprOffset = int_val(32, 0);
 
   // 3. Create the cache modifiers word
-  // bit 0: GLC = 0 (atomics drop value, less coherency)
-  // bits 1-2: SLC, DLC = 0 (similarly)
-  // bit 3: swizzled (0 for raw)
-  Value cacheModifiers = int_val(32, 0);
+  //            bit 0 = glc, bit 1 = slc, bit 2 = dlc (gfx10/gfx11),
```

Comment:
I think this could be improved:
- This is the perfect place to explain what those names are (glc, slc, dlc, swc, scc, nt, sc1, sc0, scope). 
- gfx940 and MI300 I suppose are the same thing, right? 
- Why we have the table only for MI300? Does it mean that the `switch` only applies to MI300? Would this work for say MI200?

This is the way I would structure that code. 
We support mainly 4 architectures : gfx90a, gfx942, gfx11, gfx12 . 
I don't know how different they are, but in the worst case (where all those architectures are different) I would do something like:

```
Value getCacheModifierGfx94(triton::CacheModifier cm){
// bit 0 = sc0, bit 1 = nt, bit 3 = swz, bit 4 = sc1
// Glossary:
// nt means ....
// sc0 means ...
// swz means ....
// sc1 means ....
// Translation:
  // Op   | cm  | SC1 | SC0 | NT |
  // -----+-----+-----+-----+----+--
  // Load | .ca |  0  |  0  | 0  |
  //      | .cg |  0  |  x  | 1  |
  //      | .cs |  1  |  0  | x  |
  //      | .cv |  1  |  1  | x  |
  // -----+-----+-----+-----+----+--
  // Store| .wb |  1  |  0  | 0  |
  //      | .cg |  0  |  0  | 0  |
  //      | .cs |  1  |  0  | 1  |
  //      | .wt |  1  |  1  | x  |
  int32_t aux = 0;
  switch (cm) {
  case triton::CacheModifier::CA:
    aux = 0;
    break;
  case triton::CacheModifier::CG:
    if (isBufferLoad)
      aux |= 2;
    break;
  case triton::CacheModifier::CS:
    aux |= 16;
    if (!isBufferLoad)
      aux |= 2;
    break;
  case triton::CacheModifier::CV:
    aux |= 17;
    break;
  case triton::CacheModifier::WB:
    aux |= 16;
    break;
  case triton::CacheModifier::WT:
    aux |= 17;
    break;
  default:
    aux = 0;
  }
return i32_val(aux);
}
Value getCacheModifierWordForArch(triton::CacheModifier cm, std::string arch){
switch (arch){
case "gfx90xx":
return getCacheModifierGfx90(cm);
case "gfx94xx":
return getCacheModifierGfx94(cm);
case "gfx11xx":
return getCacheModifierGfx11(cm);
case "gfx12xx":
return getCacheModifierGfx12(cm);
default:
assert(); // or return i32_val(0)
}
}
Value cacheModifiers = getCacheModifierWordForArch(targetInfo.getArch()));
```

You don't have to do that at once for all the architectures. You can start with gfx940 (which is the most important for now) and then defer the rest (i.e., the rest will return 0). Also, if two archs share the same cache modifiers, you can have a single function (and accorporate the cases). 

My broader point is that we shouldn't leave this as black magic. We want to nail down what behaviour we enforce on AMD hardware with a given Triton cache modifier. It's hard to tell for normal loads, where we are forced to obey to the backend's rules, but here we can have very fine grained control on those modifiers.

Reply:
@ramjana This is what Brian Sumner had concerns about in the "LLVM for Triton CodeGen" chat.

Reply:
I'd like to partially redact this comment. I've been reading about the difference between these cache modifiers and memory consistency which are related but not the same. I think most of my concerns are addressed with this PR only addressing BufferLoads and not AtomicLoads. Therefore we are only concerned with matching performance on a best effort basis between the PTX spec and AMD hardware and not correctness. Sorry for the confusion.

So this looks fine for gfx942.